Lib/pathlib.py

```diff
@@ -792,20 +714,32 @@ def is_absolute(self):
         a drive)."""
         if not self._root:
             return False
-        return not self._flavour.has_drv or bool(self._drv)
+        return self._flavour is posixpath or bool(self._drv)
 
     def is_reserved(self):
         """Return True if the path contains one of the special names reserved
         by the system, if any."""
-        return self._flavour.is_reserved(self._parts)
+        # FIXME: move this implementation to os.path.isreserved()
+        if self._flavour is posixpath or not self._parts:
+            return False
+
+        # NOTE: the rules for reserved names seem somewhat complicated
+        # (e.g. r"..\NUL" is reserved but not r"foo\NUL" if "foo" does not
+        # exist). We err on the side of caution and return True for paths
+        # which are not considered reserved by Windows.
+        if self._parts[0].startswith('\\\\'):
+            # UNC paths are never reserved
+            return False
+        name = self._parts[-1].partition('.')[0].partition(':')[0].rstrip(' ')
```

Comment:
I don't think you need the first partition since you are still only operating on the first `:`. Or are you relying on `IndexError` being thrown?

```suggestion
        name = self._parts[-1].partition(':')[0].rstrip(' ')
```

Reply:
These aren't equivalent:

```
>>> 'a.b'.partition('.')[0].partition(':')[0].rstrip(' ')
'a'
>>> 'a.b'.partition(':')[0].rstrip(' ')
'a.b'
```